packages/core/src/change_detection/scheduling/zoneless_scheduling_impl.ts

```diff
@@ -7,50 +7,125 @@
  */
 
 import {ApplicationRef} from '../../application/application_ref';
-import {EnvironmentProviders, inject, Injectable, makeEnvironmentProviders} from '../../di';
+import {Injectable} from '../../di/injectable';
+import {inject} from '../../di/injector_compatibility';
+import {EnvironmentProviders} from '../../di/interface/provider';
+import {makeEnvironmentProviders} from '../../di/provider_collection';
 import {PendingTasks} from '../../pending_tasks';
 import {getCallbackScheduler} from '../../util/callback_scheduler';
 import {NgZone, NoopNgZone} from '../../zone/ng_zone';
 
-import {ChangeDetectionScheduler, NotificationType} from './zoneless_scheduling';
+import {ChangeDetectionScheduler, NotificationType, ZONELESS_ENABLED, ZONELESS_SCHEDULER_DISABLED} from './zoneless_scheduling';
 
 @Injectable({providedIn: 'root'})
-class ChangeDetectionSchedulerImpl implements ChangeDetectionScheduler {
+export class ChangeDetectionSchedulerImpl implements ChangeDetectionScheduler {
   private appRef = inject(ApplicationRef);
   private taskService = inject(PendingTasks);
   private pendingRenderTaskId: number|null = null;
   private shouldRefreshViews = false;
   private readonly schedule = getCallbackScheduler();
+  private readonly ngZone = inject(NgZone);
+  private runningTick = false;
+  private cancelScheduledCallback: null|(() => void) = null;
+  private readonly zonelessEnabled = inject(ZONELESS_ENABLED);
+  private readonly disableScheduling = inject(ZONELESS_SCHEDULER_DISABLED, {optional: true});
```

Comment:
```suggestion
  private readonly disableScheduling = inject(ZONELESS_SCHEDULER_DISABLED, {optional: true}) ?? false;
```
(currently type is likely `boolean|null`)